Networking/Networking/Model/POSProduct.swift

```diff
@@ -35,6 +35,10 @@ public struct POSProduct: Codable, Equatable, GeneratedCopiable, GeneratedFakeab
         attributes.filter { $0.variation }
     }
 
+    public let manageStock: Bool
+    public let stockQuantity: Decimal?
+    public let stockStatusKey: String
```

Comment:
```suggestion
    public let stockStatusKey: String
```

Based [on documentation](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products):

> `stock_status`  Options: instock, outofstock and onbackorder.

I think we should just encode it it directly into an enum, it will be easier to work with it.

Reply:
Yeah, that's something I did on a first iteration until I saw we might expect different values that the ones shared in the API docs based on different plugins, or at least we do so in the existing stock implementation in the app side. 

We can see this in `ProductStockStatus` where they handle it with two additional cases:

```
case insufficientStock // Product Bundles only
case custom(String) // in case there are extensions modifying product stock statuses
```

At the moment I'm not sure how this will affect stock in POS, since we do not support bundles, but the custom status might do some funny business for simple products/variations. I've logged this [here](https://linear.app/a8c/issue/WOOMOB-517/woo-posproduct-labels-handle-undocumented-stock-statuses) to follow-up separately